skdownscale/pointwise_models/gard.py

```diff
@@ -36,7 +37,8 @@ def fit(self, X, y):
         self : returns an instance of self.
         """
         X, y = self._validate_data(X, y=y, y_numeric=True)
-        self.stats_ = {}  # populated in predict methods
+        self.prediction_error_ = []  # populated in predict methods
+        self.exceedance_prob_ = []  # populated in predict methods
```

Comment:
What do you think about setting these attributes in the `predict` method. Ideally, we store these values or numpy arrays (or scalars) so I'm not sure we get a whole lot of value out of creating a list here.

Reply:
(I see you've done this already below). Maybe we just remove these lines and rollback the changes in line 23?

Reply:
I think sklearn wants all attributes to be defined in `fit` first, otherwise one of the checks for sklearn compatibility fails. Are you thinking that we'd be saving some time by removing the list creation here? If so we can add to the xfail list

Reply:
Okay, I get it now. I think its okay to leave this as is for now.